cmd/gateway/commands.go

```diff
@@ -141,6 +153,15 @@ func createStaticModeCommand() *cobra.Command {
 				gwNsName = &gateway.value
 			}
 
+			var usageReportSecretNsName *types.NamespacedName
+			if cmd.Flags().Changed(usageReportSecretFlag) {
+				if !plus {
+					return errors.New("usage-report arguments are only valid if using nginx-plus")
```

Comment:
I think this is okay. If optional fields are set but not required fields, then it's a no-op, and the warning job will print in the logs. Otherwise, the required fields will have to be set together, so we only need to check one of them.